api/types/role.go

```diff
@@ -1129,6 +1151,17 @@ func (r *RoleV6) CheckAndSetDefaults() error {
 		}
 	}
 
+	for i, perm := range r.Spec.Allow.DatabasePermissions {
+		if err := perm.CheckAndSetDefaults(); err != nil {
+			return trace.BadParameter("failed to process 'allow' db_permission #%v: %v", i+1, err)
+		}
```

Comment:
Should we enforce here that wildcard is not allowed for allow db permissions?

Reply:
I'm not sure - there is already validation later in the stack. I'll give it some thought.

Reply:
Cool. i haven't reached other PRs yet. as long as it's checked somewhere 😃

Reply:
I tried to make the PRs self-contained, though, partially in order to catch issue like this one. I'll add the check.